src/python/pants/backend/helm/goals/install.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+from typing import Iterable
+
+from pants.backend.helm.target_types import HelmDeploymentFieldSet
+from pants.backend.helm.util_rules.chart import HelmChart
+from pants.backend.helm.util_rules.tool import HelmBinary, InstallOutputFormat
+from pants.core.goals.install import (
+    InstallFieldSet,
+    InstallProcess,
+    InstallProcesses,
+    InstallSubsystem,
+)
+from pants.core.util_rules.source_files import SourceFilesRequest
+from pants.core.util_rules.stripped_source_files import StrippedSourceFiles
+from pants.engine.addresses import Address
+from pants.engine.process import InteractiveProcess, InteractiveProcessRequest
+from pants.engine.rules import Get, MultiGet, collect_rules, rule
+from pants.engine.target import WrappedTarget
+from pants.engine.unions import UnionRule
+from pants.util.frozendict import FrozenDict
+from pants.util.logging import LogLevel
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class InstallHelmDeploymentFieldSet(HelmDeploymentFieldSet, InstallFieldSet):
+    pass
+
+
+_VALID_PASSTHROUGH_FLAGS = [
+    "--atomic",
+    "--dry-run",
+    "--debug",
+    "--force",
+    "--reset-values",
+    "--reuse-values",
+    "--wait",
+    "--wait-for-jobs",
+]
+
+_VALID_PASSTHROUGH_OPTS = ["--kubeconfig"]
+
+
+@rule(desc="Run Helm install process", level=LogLevel.DEBUG)
+async def run_helm_install(
+    field_set: InstallHelmDeploymentFieldSet,
+    install: InstallSubsystem,
+    helm_binary: HelmBinary,
+) -> InstallProcesses:
+    valid_args, removed_args = _cleanup_passthrough_args(install.options.args)
+    if removed_args:
+        logger.warning(f"The following arguments are ignored: {removed_args}")
```

Comment:
not really a reason to be lenient, think the initial idea was to at least be notified of the ignored args. But it is probably better to raise an error with proper message as the reason to ignore those args is because they should be configured in the `helm_deployment` target as fields.